solana-library/spl_token.sol

```diff
@@ -176,4 +176,29 @@ library SplToken {
 
 		revert("account missing");
 	}
+
+	// This struct is the return of 'get_associated_account_data'
+	struct AssociatedAccountData {
+		address owner;
+		address mintAccount;
+		uint64 balance;
+	}
+
+	/// Fetch the owner, mint account and balance for an associated token account.
+	///
+	/// @param associatedAccount The associated token account
+	/// return struct AssociatedAccountData
+	function get_associated_account_data(address associatedAccount) public view returns (AssociatedAccountData) {
```

Comment:
nit: This works for any token account, associated or non-associated, so you could call this `get_token_account_data` and `TokenAccountData`